numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,19 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=None, *args, **kw):
+        def vectorize_decorator(pyfunc):
+            ''' ``vectorize`` with presupplied keyword arguments. '''
+            return cls(pyfunc, *args, **kw)
+
+        if pyfunc is None:
+            return vectorize_decorator
+        else:
+            return object.__new__(cls)
```

Comment:
or actually, rather, it seems python 2 lets you supply arguments to object.__new__ but it doesn't actually do anything with them.  AFAICT it has only confused a large population into writing `object.__new__(cls, *args, **kwargs)`